connect/runtime/src/main/java/org/apache/kafka/connect/runtime/TransformationChain.java

```diff
@@ -26,16 +28,26 @@
 public class TransformationChain<R extends ConnectRecord<R>> {
 
     private final List<Transformation<R>> transformations;
+    private final RetryWithToleranceOperator retryWithToleranceOperator;
 
     public TransformationChain(List<Transformation<R>> transformations) {
+        this(transformations, RetryWithToleranceOperator.NOOP_OPERATOR);
```

Comment:
nit: we seem to be adding a bunch of constructors for classes like these that seem to really only be there for tests, and don't simplify things much -- basically just like 3 or 4 lines of slightly repetitive stuff. I prefer to avoid doing this since it starts to make it unclear what the *correct* constructor is; sometimes it is unavoidable, but then at least we should add a `for testing only` or similar comment. Even better is if we can keep utilities that are intended just for testing to the test jar. Here that's a bit more annoying, so leaving `NOOP_OPERATOR` in the main jar is fine, but not ideal.

The reason this stood out to me is because it looks like we're also creating metrics for a bogus connector, but since it happens via static initialization, it's not avoidable.